lib/rack/attack/check.rb

```diff
@@ -15,6 +15,7 @@ def matched_by?(request)
         block.call(request).tap do |match|
           if match
             request.env["rack.attack.matched"] = name
+            request.env['rack.attack.match_discriminator'] = match
```

Comment:
So this might be a bit odd because `Check`'s `block` are mostly intended to return `true` or `false`, not a "discrimiator" as in the case of throttling, right?

I see how the `#track` method makes this a bit confusing... given the fact that behind the scenes it used the `Throttle` or `Check` based on keyword arguments presence...

Reply:
Hi, @grzuy ! Thank you for your feedback.

Sure, you are right!. It's kind of confusing for `Check` rule to return a non-boolean value.

Motivation for proposed changes is a possibility of initial research. Before rolling out any throttling rule to production, it's would great to estimate how this rule performs (without affecting production).

If I want to test the following rule (before turning it on):
```ruby
  Rack::Attack.throttle('req/ip', :limit => 10, :period => 1.hour) do |req|
    req.ip
  end
```
I could track it, collect any real values and decide how strictly my throttling rules should be:

```
  Rack::Attack.track('requests by ip address') do |req|
    req.ip
  end
```
This way lets me make data-driven decisions. For that purpose `Track` feature fits perfectly, I think.

What do you think, It should look like? What do you think if we refactor `Check` class a bit, rename it to `Track` to make gem internals less confusing, and give my proposal a go?

Reply:
In that case wouldn't you want to

```rb
  Rack::Attack.track('req/ip', limit: 10, period: 1.hour) do |req|
    req.ip
  end
```

so that it really uses the `Throttle` (instead of `Check`) behind the scenes, in that case getting the `match_discriminator` in the request env?

Reply:
Yep, it is. My point is to initially use a non-affecting track with the same rule. It helps to determine actual traffic depending on a discriminator without affecting production.
Then, use the valid data-drive throttle limit and period for turning on to a 'trottle'.